mail-integration-services/src/main/java/org/exoplatform/mailintegration/utils/RestEntityBuilder.java

```diff
@@ -97,34 +109,60 @@ public static final MessageRestEntity fromMessage(Message message) throws Messag
 
         MimeMultipart mimeMultipart = (MimeMultipart) message.getContent();
         try {
-          body = getTextFromMimeMultipart(mimeMultipart).replaceAll("\\n|\\r", "<br>");
+          body = getTextFromMimeMultipart(mimeMultipart);
         } catch (Exception e) {
           LOG.error("error when getting body mail content", e);
         }
+        if (!inlineAttachments.isEmpty()) {
+          body = replaceInlineAttachmentsInBody(body,inlineAttachments);
+        }
         messageRestEntity.setBody(body);
         messageRestEntity.setAttachedFiles(attachedFiles.toString());
+      } else if (contentType.toLowerCase().contains("text/plain")) {
+        body = message.getContent().toString();
+        messageRestEntity.setBody(body);
+      } else if (contentType.toLowerCase().contains("text/html")) {
+        body = message.getContent().toString();
+        messageRestEntity.setBody(body);
```

Comment:
Both blocks are identical, could you merge them in one ?

Reply:
done